iroha/src/lib.rs

```diff
@@ -170,10 +170,11 @@ impl Iroha {
         let voting_handle = task::spawn(async move {
             loop {
                 if !sumeragi.write().await.voting_in_progress().await {
+                    let pending_transactions = queue.write().await.pop_pending_transactions();
```

Comment:
Unnecessary local variable which also locks `queue` and pops transactions before `sumeragi` lock.

Reply:
@humb1t this was made to fix clippy's [eval_order_dependence](https://rust-lang.github.io/rust-clippy/master/index.html#eval_order_dependence) warning. We can use `#allow[...]` instead

Reply:
Let's use allow in this case.
To fix this warning we better need to minimize usage of locks in another PR.

Reply:
The strange thing is, that clippy gave warning for the code that was not changed. This means @vmarkushin maybe your clippy version is different from the one used in our CI pipeline, as the pipeline accepted this code...

Reply:
@eadventurous seems like it is. Btw, my version is `clippy 0.0.212 (d4092ace 2020-05-11)`